### PR TITLE
fix: dry run for superbridge

### PR DESCRIPTION
**Pull request by dimsome**

### CODE REVIEW

File: script/deploy/deploy.ts [hunk 1]
```diff
@@ -291,8 +295,9 @@ const deployControllerChainContracts = async (
       deployParams
     );
 
-    deployParams.addresses[SuperBridgeContracts.Controller] =
-      controller.address;
+    deployParams.addresses[SuperBridgeContracts.Controller] = getDryRun()
+      ? AddressZero
+      : controller.address;
```
Comment (arthcp) [inline script/deploy/deploy.ts:300]: with this pattern, the stored addresses in json file would get overwritten with zero addresses if someone runs with dryrun.

File: script/helpers/common.ts [hunk 2]
```diff
@@ -54,8 +55,9 @@ export const updateConnectorStatus = async (
         siblingConnectorAddresses[it];
       if (!itConnectorAddress) continue;
 
-      let currentConnectorStatus =
-        await bridgeContract.callStatic.validConnectors(itConnectorAddress);
+      let currentConnectorStatus = getDryRun()
+        ? false
```
Comment (tHeMaskedMan981) [inline script/helpers/common.ts:59]: this is only static call right ? why need to check dryRun here?

File: script/deploy/deploy.ts (hunk 1 above)
Comment (dimsome) [reply script/deploy/deploy.ts:300]: yup, that happened indeed. Fixed this by checking in the function for dryrun

File: script/helpers/common.ts (hunk 2 above)
Comment (dimsome) [reply script/helpers/common.ts:59]: If not handled this way, the call will log and throw an exception:

Screenshot 2024-06-12 at 12 51 41

**Review by arthcp — APPROVED** [review]

**Pull request merged**